src/logics/ArithLogic.cc

```diff
@@ -959,6 +963,30 @@ ArithLogic::getDefaultValuePTRef(const SRef sref) const
         return Logic::getDefaultValuePTRef(sref);
 }
 
+PTRef ArithLogic::removeAuxVars(PTRef tr) {
+    // Note: Since ites are removed first and div/mod's then, it is important to first reintroduce div/mod's and then ites
+    class AuxSymbolMatcherConfig : DefaultRewriterConfig {
+        ArithLogic & logic;
+    public:
+        AuxSymbolMatcherConfig(ArithLogic & logic) : logic(logic) {}
+        bool previsit(PTRef) override { return true; }
+        PTRef rewrite(PTRef tr) override {
+            if (not logic.isVar(tr)) return tr; // Only variables can match
+            auto symName = std::string_view(logic.getSymName(tr));
+            if (symName.compare(0, DivModConfig::divPrefix.size(), DivModConfig::divPrefix) == 0) {
+                return DivModConfig::getDivTermFor(logic, tr);
+            } else if (symName.compare(0, DivModConfig::divPrefix.size(), DivModConfig::modPrefix) == 0){
+                return DivModConfig::getModTermFor(logic, tr);
+            }
+            return tr;
+        }
+    };
+    if (divsAndModsSeen) {
```

Comment:
Can we just *not* use this flag? It seems unnecessary to try to track this information.
Just run the rewriter everytime. We do the same in `LATheory`.
You can guard with `logic.hasIntegers()`, the same as we do in `LATheory`.

Reply:
This is a good question.  It looked like this rewriting can cause a performance bottleneck.  I didn't specifically check the effect of this optimisation though.  Checking `logic.hasIntegers()` would only help in case of `QF_LRA`.  The `QF_LIA` logic doesn't have `div`s & `mod`s, and therefore all the smtlib benchmarks would have this extra traversal for no reason.

One solution could be to record this in `ArithProperty`...

Reply:
I would be surprised if this one additional traversal would be the bottleneck.
Which is not to say that that is not case, but I would like to see some evidence.

Reply:
It feels wrong not to handle also the div&mod case, and I don't want to see this bug showing up again.